### PR TITLE
Add Collection#tap

### PR DESCRIPTION
**Pull request by 1Computer1**
**Please describe the changes this PR makes and why it should be merged:**

Adds ~~`Collection#inspect`~~ `Collection#tap` which allows for easy debugging when chaining collection methods. 
Naming comes from ~~Rust~~ Laravel.

**Semantic versioning classification:**  
- [X] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/util/Collection.js [hunk 1]
```diff
@@ -364,6 +364,24 @@ class Collection extends Map {
...
+   * [Map.forEach()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/forEach),
+   * but returns the collection instead of undefined.
+   * @param {Function} fn Function to execute for each element
+   * @param {*} [thisArg] Value to use as `this` when executing function
+   * @returns {Collection}
+   * @example
+   * collection
+   *  .inspect(user => console.log(user.username))
+   *  .filter(user => user.bot)
+   *  .inspect(u => console.log(user.username))
```
Comment (kyranet) [inline src/util/Collection.js]: `user` is undefined in this scope, the variable is defined as `u` here.

**Comment by devsnek** [comment]
this cannot be named inspect due to how node's util.inspect method works

**Comment by 1Computer1** [comment]
Suggestions for name?

**Comment by rei2hu** [comment]
What about `examine`?

**Comment by devsnek** [comment]
why don't you just `return this` in our existing forEach method.............

**Comment by 1Computer1** [comment]
1. there isn't one (that comes from Map)
2. would then change the interface of Map and i dunno if people like that

**Comment by devsnek** [comment]
how do you `return this.forEach` if there we didn't implement it 🤔

**Comment by devsnek** [comment]
he'll it's called forEach just name it forEach why are you trying to make the name as cryptic as possible

**Comment by gloss-water** [comment]
name it snekEach

**Comment by lloti** [comment]
I agree with Gus on this one. You should call it forEach and call super.forEach <Map>.forEach returns undefined by default anyways so I doubt it'll break anyone's code.

**Pull request merged**